expression/aggregation/util.go

```diff
@@ -54,7 +55,7 @@ func (d *distinctChecker) Check(values []types.Datum) (bool, error) {
 }
 
 // calculateSum adds v to sum.
-func calculateSum(sc *stmtctx.StatementContext, sum, v types.Datum) (data types.Datum, err error) {
+func calculateSum(sc *stmtctx.StatementContext, sum, v types.Datum, retType *types.FieldType) (data types.Datum, err error) {
```

Comment:
What's the relationship between pushdown CAST and `calculateSum`? Are there any test cases for `sum`?

Reply:
There are many test cases that will cover the `sum` (query with `sum/avg` aggregation functions). The is no direct relationship with `CAST` and `sum` and I modify it because of the previous implementation does not round appropriately (this BUG is detected by this PR).